index.bs

```diff
@@ -391,6 +391,54 @@ Implementers of this API are expected to be familiar with the <a href="https://g
 
 
 # Programming Model # {#programming-model}
+## Overview ## {#programming-model-overview}
+
+At the heart of neural networks is a computational graph of mathematical operations.
+These operations are the building blocks of modern machine learning technologies in
+computer vision, natural language processing, and robotics.
+The WebNN API is a specification for constructing, compiling, and executing computational
+graphs of neural networks.
+
+The {{MLGraph}} interface represents a compiled computational graph (that is, a model) and exposes
+a compute method to perform inference.
+
+The {{MLGraphBuilder}} interface serves as a builder (factory) to create a {{MLGraph}}.
+(In programming language terminology, a {{MLGraph}} represents a *program/function*.)
+The {{MLOperand}} interface is a static (compile-time) representation of runtime values,
+which include input-values for inference, constants (including trained weights)
+used for inference, intermediate values (often referred to as activations) computed
+during inference, as well as the output values of inference.
```

Comment:
Thanks for updating the wordings. I just think that it's a lot simpler to explain `MLOperand` as a *description* of the data to be processed by the graph at execution time, and not the data itself aka the *tensors*. Hopefully that saves you from having to invoke the *in programming language terminology..* paradigm, which is rather conceptual in my view.

Reply:
Thanks Chai for the suggestion. The problem I have is drawing yet another distinction: A _tensor_ itself  arguably consists of _data_ and a description of the data (such as shape, for example). So, I find the terms "data" and "description of data" themselves to be ambiguous (with these two different interpretations). I couldn't find a simple way to summarize this unambiguously. The key distinguishing feature of an MLOperand is that this is a description that is available prior to execution-time (that is, a static or compile-time description). I realize that I used the word "represention" above. Would it help to change it to ```static (compile-time) description of runtime data```?

Reply:
@wchao1115 
> `MLOperand` as a _description_ of the data to be processed by the graph at execution time, and not the data itself aka the _tensors_

This is not always true. For [constant `MLOperand`](https://webmachinelearning.github.io/webnn/#dom-mlgraphbuilder-constant), it contains both description and data.

@gramalingam
> I realize that I used the word "representation" above.

I think "presentation" is good. To me, the computation graph contains the tensor operations. The `MLOperand` just represents the tensor of operation when building the computational graph. They could represent input, output, constant and intermediate tensor. At compute-time, the caller should bind the value to the input `MLOperand` and provide the pre-allocated buffer to the output `MLOperand` that receives the value.

Reply:
> For constant MLOperand, it contains both description and data.

True.

> I couldn't find a simple way to summarize this unambiguously.

What if we just focus on explaining the nature of an operand as a representation of the data that flows within the computational graph including the constant, input, output, and intermediate data between two operations? I feel that by describing it directly this way, the reference to the programming language terminology could be set aside.

Reply:
> What if we just focus on explaining the nature of an operand as a representation of the data that flows within the computational graph including the constant, input, output, and intermediate data between two operations?

sgtm

Reply:
Sounds good to me too. I updated it accordingly. Please take a look. Thanks!